crates/query-engine/sql/src/sql/ast.rs

```diff
@@ -225,10 +215,37 @@ pub enum BinaryArrayOperator {
 
 /// A scalar function
 #[derive(Debug, Clone, PartialEq, Eq)]
-pub enum Function {
```

Comment:
We now have all sorts of functions beyond `Coalesce` and `JsonAgg`, and it doesn't seem to buy us much to enumerate all the functions we care about when we're just cramming them into an AST, so `Function` is now a struct.